defaultconfig/template.conf

```diff
@@ -0,0 +1,196 @@
+    application/vnd.wap.wmlc wmlc;
+    application/vnd.google-earth.kml+xml  kml;
+    application/vnd.google-earth.kmz kmz;
+    application/x-7z-compressed 7z;
+    application/x-cocoa cco;
+    application/x-java-archive-diff jardiff;
+    application/x-java-jnlp-file jnlp;
+    application/x-makeself run;
+    application/x-perl pl pm;
+    application/x-pilot prc pdb;
+    application/x-rar-compressed rar;
+    application/x-redhat-package-manager  rpm;
+    application/x-sea sea;
+    application/x-shockwave-flash swf;
+    application/x-stuffit sit;
+    application/x-tcl tcl tk;
+    application/x-x509-ca-cert der pem crt;
+    application/x-xpinstall xpi;
+    application/xhtml+xml xhtml;
+    application/zip zip;
+    application/octet-stream bin exe dll;
+    application/octet-stream deb;
+    application/octet-stream dmg;
+    application/octet-stream eot;
+    application/octet-stream iso img;
+    application/octet-stream msi msp msm;
+    application/json json;
+    audio/midi mid midi kar;
+    audio/mpeg mp3;
+    audio/ogg ogg;
+    audio/x-m4a m4a;
+    audio/x-realaudio ra;
+    video/3gpp 3gpp 3gp;
+    video/mp4 mp4;
+    video/mpeg mpeg mpg;
+    video/quicktime mov;
+    video/webm webm;
+    video/x-flv flv;
+    video/x-m4v m4v;
+    video/x-mng mng;
+    video/x-ms-asf asx asf;
+    video/x-ms-wmv wmv;
+    video/x-msvideo avi;
+  }
+
+  # TODO: Rename log format? Leave as is?
+  log_format paketo '$http_x_forwarded_for - $http_referer - [$time_local] "$request" $status $body_bytes_sent';
+  # TODO: write logs to /tmp?
+  access_log {{ env "APP_ROOT" }}/logs/access.log paketo;
+
+  # Set the default MIME type of responses; 'application/octet-stream'
+  # represents an arbitrary byte stream
+  default_type application/octet-stream;
+
+  # (Performance) When sending files, skip copying into buffer before sending.
+  sendfile on;
+  # (Only active with sendfile on) wait for packets to reach max size before
+  # sending.
+  tcp_nopush on;
+
+  # (Performance) Enable compressing responses
+  gzip on;
+  # For all clients
+  gzip_static always;
+  # Including responses to proxied requests
+  gzip_proxied any;
+  # For responses above a certain length
+  gzip_min_length 1100;
+  # That are one of the following MIME types
+  gzip_types text/plain text/css text/js text/xml text/javascript application/javascript application/x-javascript application/json application/xml application/xml+rss;
+  # Compress responses to a medium degree
+  gzip_comp_level 6;
+  # Using 16 buffers of 8k bytes each
+  gzip_buffers 16 8k;
+
+  # Add "Vary: Accept-Encoding” response header to compressed responses
+  gzip_vary on;
+
+  # Decompress responses if client doesn't support compressed
+  gunzip on;
+
+  # Don't compress responses if client is Internet Explorer 6
+  gzip_disable "msie6";
+
+  # Set a timeout during which a keep-alive client connection will stay open on
+  # the server side
+  # TODO: necessary? vv
+  keepalive_timeout 30;
```

Comment:
For now, leaving at 30 seconds since this is what's set in the staticfile buildpack and I imagine many users will be migrating from that. We can always tweak it later.